### PR TITLE
boot, o/devicestate: TrustedAssetUpdateObserver stubs, hook up to gadget updates

### PR DESCRIPTION
**Pull request by bboozzoo**
The branch introduces a TrustedAssetUpdateObserver that shall track the trusted assets getting updated during gadget assets update and attempt a reseal when needed.

The bits in overlord/devicestate hook up the right observer to the update process.

### CODE REVIEW

File: boot/assets.go [hunk 1]
```diff
@@ -63,3 +60,48 @@ func (o *TrustedAssetsInstallObserver) Seal() error {
 	// - initial seal
 	return nil
 }
+
+func NewTrustedAssetsUpdateObserver(model *asserts.Model) *TrustedAssetsUpdateObserver {
```
Comment (mvo5) [inline boot/assets.go:64]: (nitpick) This feels slightly iffy as NewXXX() usually creates the thing. Maybe `TrustedAssetsUpdateObserverForModel(model)` or something?

Comment (pedronis) [reply boot/assets.go:64]: yes, @mvo5 is right, I missed this in the other PR, but returning a nil from New is very unusual, this should probably return an error in the other case, with the doc comment explaining what's the only possible error case.

NewTrustedAssetsInstallObserver needs changing as well

File: boot/assets.go [hunk 2]
```diff
@@ -63,3 +60,48 @@ func (o *TrustedAssetsInstallObserver) Seal() error {
...
+	// TODO:UC20:
+	// - reseal with a given state of modeenv
+	return nil
+}
+
+// Canceled is called when the update has been canceled, or if changes
+// were written and the update has been reverted.
+func (o *TrustedAssetsUpdateObserver) Canceled() error {
+	// TODO:UC20:
+	// - reseal with a given state of modeenv
```
Comment (pedronis) [inline boot/assets.go:105]: it should drop assets as well, no?

File: boot/assets.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ import (
 	"github.com/snapcore/snapd/gadget"
 )
 
-func NewTrustedAssetsInstallObserver(model *asserts.Model) *TrustedAssetsInstallObserver {
+func TrustedAssetsInstallObserverForModel(model *asserts.Model) *TrustedAssetsInstallObserver {
```
Comment (pedronis) [inline boot/assets.go:27]: this needs a doc comment, especially because of the return nil behavior

File: boot/assets.go [hunk 4]
```diff
@@ -63,3 +60,49 @@ func (o *TrustedAssetsInstallObserver) Seal() error {
 	// - initial seal
 	return nil
 }
+
+func TrustedAssetsUpdateObserverForModel(model *asserts.Model) *TrustedAssetsUpdateObserver {
```
Comment (pedronis) [inline boot/assets.go:64]: same

**Pull request merged**